services/ui-src/src/components/fields/TextField.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect } from "react";
+import { useFormContext } from "react-hook-form";
+import { TextField as CmsdsTextField } from "@cmsgov/design-system";
+import { Box } from "@chakra-ui/react";
+import { parseCustomHtml } from "utils";
+import { TextboxTemplate } from "../../types/report";
+import { PageElementProps } from "../report/Elements";
+
+export const TextField = (props: PageElementProps) => {
+  const textbox = props.element as TextboxTemplate;
+  const defaultValue = "";
+  const [displayValue, setDisplayValue] = useState<string>(defaultValue);
+
+  // get form context and register field
+  const form = useFormContext();
+  const inputName = "answers.1.2"; // TODO: id based
+
+  useEffect(() => {
+    form.register(inputName);
+  }, []);
+
+  const onChangeHandler = async (
```

Comment:
Does this work with onBlur only or does it need onChange? My concern is frequency of updates being shuffled up to api calls

Reply:
my understanding was that both are eventually needed (if they click next without blurring from the field)

Reply:
I think clicking Next _will_ blur from the field, and `onChange` fires every keystroke.

If we want field validation on keystroke, then we need an onChange handler. If we want to limit API calls, we should only send data back to the API onBlur.

In our other apps, onBlur is when we would call autosave code. If HCBS automatically percolates store updates to the API, and we're sending data to the store on every keystroke, maybe we could do some kind of debounce?

Alternatively, perhaps we could divorce store updates from field validation somehow? I'm not sure how that would work, though...